pkg/common/utils.go

```diff
@@ -479,12 +473,43 @@ func existingErrorCode(err error) (codes.Code, error) {
 	if err == nil {
 		return codes.Unknown, fmt.Errorf("null error")
 	}
-	if status, ok := status.FromError(err); ok {
-		return status.Code(), nil
+	var tmpError *TemporaryError
+	if errors.As(err, &tmpError) {
+		if status, ok := status.FromError(err); ok {
+			return status.Code(), nil
+		}
+	}
+	var googleErr *googleapi.Error
+	if !errors.As(err, &googleErr) {
+		if status, ok := status.FromError(err); ok {
+			return status.Code(), nil
+		}
 	}
+
 	return codes.Unknown, fmt.Errorf("no existing error code for %w", err)
 }
 
+func isAPIError(err error) (codes.Code, error) {
+	var googleErr *googleapi.Error
+	if errors.As(err, &googleErr) {
```

Comment:
nit: this entire function is in an if statement which checks that the error is a googleAPI error. We could do 
```
	var googleErr *googleapi.Error
	if !errors.As(err, &googleErr) {
		return codes.Internal
	}
```
first then return, to reduce nested if statments.